### PR TITLE
Add simple docs on how to build things

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
I mostly wanted to have my build commands written down. But should be useful.

### CODE REVIEW

File: docs/developer/build_and_test.html [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+ -->
+
+ <meta charset="utf-8" lang="en"><style class="fallback">body{visibility:hidden;}</style>
+ **Expressive Vector Engine**
+
+(insert ../crumbs.html here)
+
+# Building EVE: how to do local development.
+
+*NOTE: this documentation was valid at [commit](https://github.com/jfalcou/eve/commit/2fe1048aa31eb3271666f1cca04bf2f5f34eeb6e) and might get outdated.* <br/>
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:18]: as any file, I guess this kind of disclaimer is not useful.

File: docs/developer/build_and_test.html [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve
+  mkdir build_avx512 && cd build_avx512
+  cmake -G Ninja .. '-DCMAKE_CXX_FLAGS=-march=skylake-avx512 -DEVE_NO_FORCEINLINE'
+  ninja -j 16 unit.real.algorithm.exe
+  sde -- ./unit/unit.real.algorithm.all.regular.simd.exe
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+ctest doesn't work with sde for me and I didn't need it yet.
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:162]: This is where we should talk about the CMAKE_CROSSCOMPILING_CMD option

File: docs/developer/build_and_test.html [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  ninja -j 16 unit.meta.exe
+  ../cmake/toolchain/run_arm.sh ./unit/unit.meta.as_integer.exe
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+## AVX-512 Docker Example
+
+Now, avx-512 is not supported on my machine directly, so I use sde to run it.
+It is supported in our build envirenment so we do not use it for CI.
+I also use the EVE_OPTIONS override to march.
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:129]: s/I/We

File: docs/developer/build_and_test.html [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  cmake -G Ninja .. -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain/gcc.arm.cmake
+  ninja -j 16 unit.meta.exe
+  ../cmake/toolchain/run_arm.sh ./unit/unit.meta.as_integer.exe
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+## AVX-512 Docker Example
+
+Now, avx-512 is not supported on my machine directly, so I use sde to run it.
+It is supported in our build envirenment so we do not use it for CI.
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:128]: spurrious details

File: docs/developer/build_and_test.html [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+How we build for certain platforms can be found in: [config](https://github.com/jfalcou/eve/blob/2fe1048aa31eb3271666f1cca04bf2f5f34eeb6e/.circleci/config.yml#L14)<br/>
+Toolchains suppresses the default CMAKE compiler options, so we have EVE_OPTIONS macro you can define instead.<br/>
+For example for different x86 we use the same toolchain, so you need to use EVE_OPTIONS to specify the march.
+
+## Arm v7 Docker Example.
+
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve
+  # in docker to the left I see root@<commit>:
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:117]: We can actually just copy/paste an actual output

File: docs/developer/build_and_test.html [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+  # pwd eve
+  dockhere jfalcou/compilers:latest
+  # cd where eve is again
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+5: Now you can use all of the EVE's prepackaged compilers.
+
+You probably want to use our toolchain files for it.
+How we build for certain platforms can be found in: [config](https://github.com/jfalcou/eve/blob/2fe1048aa31eb3271666f1cca04bf2f5f34eeb6e/.circleci/config.yml#L14)<br/>
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:108]: We can make a table so for each Arch we display the toolchain + option as hints

File: docs/developer/build_and_test.html [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  sudo snap install docker
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+2: Make docker work.
+
+Get to the point when "docker run hello-world" is succesfull.
+Google solved my problems reasonably fast
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:80]: Hahaha 'put under the carpet' award of the year ^^
I think point 1 and 2. can be merged into a "Here is a good tutorial on settign up docker, you should follow tseps up to the poitn where .... works".

File: docs/developer/build_and_test.html [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve/build
+  time ninja -j 16 unit.exe && ctest -j 16
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+This will run all of the EVE unittests. Which maybe a bit much, depending on what you do.
+
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:55]: There we should explain the test naming scheme and target network. I an write that if you want

File: docs/developer/build_and_test.html [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve
+  mkdir build && cd build
+  cmake -G Ninja .. '-DCMAKE_CXX_FLAGS=-march=native -g -DEVE_NO_FORCEINLINE' -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+The only non standard option here is -DEVE_NO_FORCEINLINE. EVE force inlines aggressively
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:40]: This is to be made a note.

File: docs/developer/build_and_test.html [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+  mkdir build_avx512 && cd build_avx512
+  cmake -G Ninja .. -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchain/clang.x86.cmake -DEVE_OPTIONS="-march=skylake-avx512"
+  ninja -j 16 unit.meta.exe
+  sde -- ./unit/unit.meta.as_integer.exe
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+## Other saved examples
+
+I prefer to not work from docker when I don't have to.
```
Comment (jfalcou) [inline docs/developer/build_and_test.html:144]: s/I/We

**Review by jfalcou — CHANGES_REQUESTED** [review]
I think the contents is OK but:
 - we need to work on the tone first/3rd person style so it feels like an actual doc page more than a blog page
 - We need a clear separation between building locally, building in the Docker. I think we should start with building locally using g++ or/and Clang, setting up options, and getting a grasp of the test target naming schemes. Then, we introduce the docker and how you can run an arbitrary compiler INSIDE the docker. If needed, I can write the part where we tell the people how to call the Docker compiler from outside the docker.
 - For the bash example, just copy paster w/e appears in your actual bash.
 - We need to explain the CMAKE_CROSSCOMPILING_CMD option at some point

File: docs/developer/build_and_test.html (hunk 1 above)
Comment (DenisYaroshevskiy) [reply docs/developer/build_and_test.html]: I think this is a good practice in docs.

**Pull request merged**